scripts/fastlane/Fastfile

```diff
@@ -0,0 +1,192 @@
+  version_regex = /(\d*)\.(\d*)\.(\d*)(?:-unstable\.)?(\d*)?/
+  pr_body = %Q(
+## Release PR review checklist
+- [ ] Verify version name in gradle.properties
+- [ ] Verify version code in gradle.properties
+- [ ] Verify CHANGELOG.md
+)
+
+  desc "Fetch all tags to ensure accurate calculation of the next release number."
+  before_all do 
+    sh('git', 'fetch', '--tags')
+    sh('git', 'fetch')
+  end
+
+  lane :configure_git_options do |options|
+    sh('git', 'config', 'user.email', options[:git_user_email])
+    sh('git', 'config', 'user.name', options[:git_user_name])
+  end
+
+  lane :manage_next_release_pr do |options|
+    product_name = options[:product_name]
+    last_release_tag = last_git_tag(pattern: "#{RELEASE_TAG_PREFIX}*")
+    next_version, commits = calculate_next_release_version(release_tag_prefix: RELEASE_TAG_PREFIX, from_tag: last_release_tag)
+    UI.message("Version change #{last_release_tag} => #{next_version}")
+    update_gradle_properties(version: next_version)
+    update_change_log(last_release_tag: last_release_tag)
+
+    git_add(path: [gradle_properties_path, change_log_path])
+    git_commit(path: [gradle_properties_path, change_log_path], message: "chore(release): #{product_name} - #{next_version}")
+    push_to_git_remote(force: true)
+
+
+    current_branch = sh('git', 'branch', '--show-current')
+    repo_name_parts = options[:repo].split('/')
+    create_pull_request(base: 'main', 
+                        title: "chore(release): #{product_name} - #{next_version}", 
+                        body:pr_body, 
+                        repo: options[:repo], 
+                        api_token: ENV["RELEASE_MANAGER_TOKEN"])
+  end
+
+
+  desc "Checks out a tag and runs the uploadArchives gradle task."
+  lane :push_tag_to_maven do |options|
+    UI.message("Pushing release #{options[:release_tag]} to maven")
+    sh("git", "checkout", options[:release_tag])
+    publish_to_maven
+  end
+
+  desc "Update changelog"
+  private_lane :update_change_log do |options|
+    changelog_body = changelog_from_git_commits(
+      between: [options[:last_release_tag], "HEAD"],
+      merge_commit_filtering: "exclude_merges",
+      pretty: "- %s"
+    )
+
+    open(change_log_path, 'a') { |f|
+      f.puts "# Release Amplify Android #{options[:next_version]}\n\n"
+      f.puts "#{changelog_body}\n\n"
+    }
+  end
+
+  desc "Increment versions"
+  private_lane :update_gradle_properties do |options|
+    version = options[:version].to_s
+    segments = version.match(version_regex).captures # version.split('.')
+    version_code = calculate_version_code(segments: segments, mode: 3)
+
+    UI.message("Updating versionName and versionCode in gradle.properties")
+
+    set_key_value(file: gradle_properties_path, key: 'VERSION_NAME', value: version)
+    set_key_value(file: gradle_properties_path, key: 'VERSION_CODE', value: version_code.to_s)
+
+  end
+
+  # Calculate the Android version code with one of 3 different approaches.
```

Comment:
I never found anything authoritative on that. When discussing versioning in the developer guide, they don't even mention libraries. Only thing I found was a [Stackoverflow post](https://stackoverflow.com/questions/57230308/do-i-need-to-increase-version-code-of-a-library-module).

My reasoning for exploring multiple options was mainly related to trying to find a simpler way to come up with that number without creating constraints. 

Given the fact that's it's probably not even used, I'll clean up some of these other options and leave option 1 in there just in case. We can try to do a release without incrementing it and see what happens.